model_compression_toolkit/core/common/mixed_precision/sensitivity_evaluation_manager.py

```diff
@@ -137,34 +154,32 @@ def build_distance_metrix(self):
         return distance_matrix
 
 
-def get_mp_interest_points(graph: Graph, fw_info: FrameworkInfo, num_ip_factor: float) -> List[BaseNode]:
+def get_mp_interest_points(graph: Graph, interest_points_classifier: Callable, num_ip_factor: float) -> List[BaseNode]:
     """
     Gets a list of interest points for the mixed precision metric computation.
     The list is constructed from a filtered set of the convolutions nodes in the graph.
 
     Args:
         graph: Graph to search for its MP configuration.
-        fw_info: FrameworkInfo object about the specific framework
-                (e.g., attributes of different layers' weights to quantize).
+        interest_points_classifier: A function that indicates whether a given node in considered as a potential
+            interest point for mp metric computation purposes.
         num_ip_factor: Percentage out of the total set of interest points that we want to actually use.
 
     Returns: A list of interest points (nodes in the graph).
 
     """
     sorted_nodes = graph.get_topo_sorted_nodes()
-    kernel_nodes = list(filter(lambda n: fw_info.get_kernel_op_attributes(n.type)[0] is not None, sorted_nodes))
+    ip_nodes = list(filter(lambda n: interest_points_classifier(n), sorted_nodes))
 
-    interest_points_nodes = bound_num_interest_points(kernel_nodes, num_ip_factor)
+    interest_points_nodes = bound_num_interest_points(ip_nodes, num_ip_factor)
 
-    # We add the last configurable node to the list of interest points, since we need it to be able to include all
-    # configurable nodes in the sensitivity metric computation
+    # We add last layer in model to interest points,
+    # in order to consider the model's output in the distance metric computation (and also to make sure
+    # all configurable layers are included in the configured mp model for metric computation purposes)
+    last_layer_node = sorted_nodes[-1]
+    interest_points = interest_points_nodes if last_layer_node in interest_points_nodes \
+        else interest_points_nodes + [last_layer_node]
 
```

Comment:
I think we should take all outputs instead of just the last node in the sorted nodes